### PR TITLE
[17.0][MIG] base_external_dbsource

### PR DESCRIPTION
**Pull request by Anxo82**
Standard migration

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check pre-commit: https://github.com/OCA/maintainer-tools/wiki/Install-pre-commit

/ocabot migration base_external_dbsource

File: base_external_dbsource/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright <2011> <Daniel Reis, Maxime Chambreuil, Savoir-faire Linux>
+# Copyright <2016> <Henry Zhou MAXodoo>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "External Database Sources",
```
Comment (carolinafernandez-tecnativa) [inline base_external_dbsource/__manifest__.py:5]: Version should be 17.0.1.0.0

File: base_external_dbsource/demo/base_external_dbsource.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" ?>
+<odoo>
```
Comment (carolinafernandez-tecnativa) [inline base_external_dbsource/demo/base_external_dbsource.xml]: Please add noupdate="1"

File: base_external_dbsource/tests/test_base_external_dbsource.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+from unittest import mock
+
+from odoo.sql_db import connection_info_for
+from odoo.tests import common
+
+
+class TestBaseExternalDbsource(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
```
Comment (carolinafernandez-tecnativa) [inline base_external_dbsource/tests/test_base_external_dbsource.py]: Please use setUpClass instead

**Review by carolinafernandez-tecnativa — COMMENTED** [review]
Plase Squash administrative commits. Check https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate for details.

**Comment by sergio-teruel** [comment]
Hi @dalonsod 

Can you fix the main view??

image

File: base_external_dbsource/views/base_external_dbsource.xml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                            />
+                            <button
+                                name="connection_test"
+                                string="Test Connection"
+                                type="object"
+                                icon="fa-refresh"
+                            />
+                        </group>
+                    </group>
+                </sheet>
```
Comment (sergio-teruel) [inline base_external_dbsource/views/base_external_dbsource.xml]:
```suggestion
                <sheet>
                    <group>
                        <group>
                            <field name="name"/>
                            <field
                                    name="company_id"
                                    groups="base.group_multi_company"
                                    options="{'no_create': True}"
                            />
                        </group>
                        <group>
                            <field name="password" password="True" />
                        </group>
                    </group>
                    <group>
                        <field name="connector" />

                    </group>
                    <group string="Connection String">
                        <field
                                name="conn_string"
                                nolabel="1"
                                placeholder="Please check the tooltip for connection string examples"
                            />
                        <group/>
                    </group>
                </sheet>```

File: base_external_dbsource/tests/test_base_external_dbsource.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
+# Copyright 2016 LasLabs Inc.
+
+from unittest import mock
+
+from odoo.sql_db import connection_info_for
+from odoo.tests import common
+
+
+class TestBaseExternalDbsource(common.TransactionCase):
+    def setUpClass(self):
```
Comment (pedrobaeza) [inline base_external_dbsource/tests/test_base_external_dbsource.py]: You must adapt this method, as the argument is `cls`, not `self`, and you are also missing the `@classmethod` decorator. Please check other modules with this configuration. This is done for optimizing the time of the tests, doing the setup only once, not once per test.

**Comment by carolinafernandez-tecnativa** [comment]
Plase Squash administrative commits. Check https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate for details.

**Comment by carolinafernandez-tecnativa** [comment]
Hi, could you please check my previous comment to squash administrative commits so can we move forward? Thanks!

File: base_external_dbsource/demo/base_external_dbsource.xml (hunk 2 above)
Comment (pedrobaeza) [reply base_external_dbsource/demo/base_external_dbsource.xml]: This comment is pending

File: base_external_dbsource/models/base_external_dbsource.py [hunk 6]
```diff
@@ -322,7 +322,7 @@ def _get_adapter_method(self, method_prefix):
         """
 
         self.ensure_one()
-        method = "{}_{}".format(method_prefix, self.connector)
+        method = f"{method_prefix}_{self.connector}"
```
Comment (carolinafernandez-tecnativa) [inline base_external_dbsource/models/base_external_dbsource.py:325]: This change must be done on migration commit, not in this commit.

File: base_external_dbsource/demo/base_external_dbsource.xml (hunk 2 above)
Comment (carolinafernandez-tecnativa) [reply base_external_dbsource/demo/base_external_dbsource.xml]: This comment is pending

**Review by carolinafernandez-tecnativa — COMMENTED** [review]
Please squash administrative commits and take into account the comments pendings and after that, we could proceed to migrate.
Thanks!

File: base_external_dbsource/models/base_external_dbsource.py (hunk 6 above)
Comment (Anxo82) [reply base_external_dbsource/models/base_external_dbsource.py:325]: Not sure, but this change was added by `pre-commit` during pre-commit auto fixes stuff.

File: base_external_dbsource/demo/base_external_dbsource.xml (hunk 2 above)
Comment (Anxo82) [reply base_external_dbsource/demo/base_external_dbsource.xml]: Comment attended

**Comment by Anxo82** [comment]
> Please squash administrative commits and take into account the comments pendings and after that, we could proceed to migrate.
> Thanks!

Clean commit history done

**Review by carolinafernandez-tecnativa — APPROVED** [review]
LGTM Thanks :) ping @sergio-teruel

**Review by pedrobaeza — APPROVED** [review]
There's still some changes in the README that doesn't belong to the pre-commit auto fixes, but let's unblock this.

/ocabot merge nobump

**Pull request merged**